### ISSUE TITLE
Unknown Server error from http://127.0.0.1:8000

### ISSUE THREAD

**Issue by jean-christophe-manciot** [GNS3/gns3-server#914]
- Updated Ubuntu 16.10
- GNS3 GUI 1.5.3.1
- GNS3 Server 1.5.3.1
- GNS3 Ubridge 0.9.11
- qemu 2.8.0
- running node: IOSv-L2 15.2(4.0.55)E
- nodes which cannot start: NX-OSv 7.3.0

After some time (~ 5mn), it is impossible to start any qemu node once at least one other qemu node is running.
In gns3_gui.log:
```
...
2017-02-24 18:58:42 INFO http_client.py:682 Response error: Unknown error (error: 99)
2017-02-24 18:58:42 INFO http_client.py:271 Connection to http://127.0.0.1:8000 closed
2017-02-24 18:58:42 ERROR vm.py:146 error while starting NX_OSv_Spine_11: Unknown error
```
In gns3_server.log:
```
...
2017-02-24 18:54:31 INFO helpers.py:473 127.0.0.1 - - [24/Feb/2017:17:54:31 +0000] "POST /v1/projects/d3f1b32d-60af-4481-b8d7-66a875013624/qemu/vms/f3273ad8-cac8-4a45-bc6a-6d17aa7d677f/start HTTP/1.1" 200 2731 "-" "GNS3 QT Client v1.5.4dev1"
2017-02-24 18:55:41 INFO helpers.py:473 127.0.0.1 - - [24/Feb/2017:17:55:41 +0000] "-" 504 182 "(no headers)" "(no headers)"
2017-02-24 18:55:41 INFO helpers.py:473 127.0.0.1 - - [24/Feb/2017:17:55:41 +0000] "-" 504 182 "(no headers)" "(no headers)"
2017-02-24 18:55:41 INFO helpers.py:473 127.0.0.1 - - [24/Feb/2017:17:55:41 +0000] "-" 504 182 "(no headers)" "(no headers)"
2017-02-24 18:55:41 INFO helpers.py:473 127.0.0.1 - - [24/Feb/2017:17:55:41 +0000] "-" 504 182 "(no headers)" "(no headers)"
2017-02-24 18:55:45 INFO helpers.py:473 127.0.0.1 - - [24/Feb/2017:17:55:45 +0000] "-" 504 182 "(no headers)" "(no headers)"
```
On the target node which cannot be started:
```
Start QEMU with /usr/bin/qemu-system-x86_64 -name NX_OSv_Spine_11 -m 3072M -smp cpus=1 -enable-kvm -machine smm=off -boot order=c -drive file=/media/actionmystique/SAMSUNG-850-Ext4/Labs/GNS3/CCNP-Cloud/CLDAUT/Cisco-Nexus-Cloud/7.3.0/project-files/qemu/6ad53673-0a8a-4845-b3a5-e6d3fe1e9e56/hda_disk.qcow2,if=virtio,index=0,media=disk -serial telnet:127.0.0.1:2001,server,nowait -monitor tcp:127.0.0.1:42699,server,nowait -net none -device virtio-net-pci,mac=00:00:ab:9e:56:00,netdev=gns3-0 -netdev socket,id=gns3-0,udp=127.0.0.1:15007,localaddr=127.0.0.1:15010 -device virtio-net-pci,mac=00:00:ab:9e:56:01 -device virtio-net-pci,mac=00:00:ab:9e:56:02 -device virtio-net-pci,mac=00:00:ab:9e:56:03,netdev=gns3-3 -netdev socket,id=gns3-3,udp=127.0.0.1:15000,localaddr=127.0.0.1:15002 -device virtio-net-pci,mac=00:00:ab:9e:56:04,netdev=gns3-4 -netdev socket,id=gns3-4,udp=127.0.0.1:15008,localaddr=127.0.0.1:15006 -device virtio-net-pci,mac=00:00:ab:9e:56:05,netdev=gns3-5 -netdev socket,id=gns3-5,udp=127.0.0.1:15020,localaddr=127.0.0.1:15017 -device virtio-net-pci,mac=00:00:ab:9e:56:06,netdev=gns3-6 -netdev socket,id=gns3-6,udp=127.0.0.1:15022,localaddr=127.0.0.1:15023 -device virtio-net-pci,mac=00:00:ab:9e:56:07 -device virtio-net-pci,mac=00:00:ab:9e:56:08 -device virtio-net-pci,mac=00:00:ab:9e:56:09 -device virtio-net-pci,mac=00:00:ab:9e:56:0a -device virtio-net-pci,mac=00:00:ab:9e:56:0b -device virtio-net-pci,mac=00:00:ab:9e:56:0c -device virtio-net-pci,mac=00:00:ab:9e:56:0d -device virtio-net-pci,mac=00:00:ab:9e:56:0e -device virtio-net-pci,mac=00:00:ab:9e:56:0f -device virtio-net-pci,mac=00:00:ab:9e:56:10 -device virtio-net-pci,mac=00:00:ab:9e:56:11 -device virtio-net-pci,mac=00:00:ab:9e:56:12 -device virtio-net-pci,mac=00:00:ab:9e:56:13 -device virtio-net-pci,mac=00:00:ab:9e:56:14 -device virtio-net-pci,mac=00:00:ab:9e:56:15 -device virtio-net-pci,mac=00:00:ab:9e:56:16 -device virtio-net-pci,mac=00:00:ab:9e:56:17 -nographic -enable-kvm -cpu host

Execution log:
Warning: nic virtio-net-pci.1 has no peer
Warning: nic virtio-net-pci.2 has no peer
Warning: nic virtio-net-pci.7 has no peer
Warning: nic virtio-net-pci.8 has no peer
Warning: nic virtio-net-pci.9 has no peer
Warning: nic virtio-net-pci.10 has no peer
Warning: nic virtio-net-pci.11 has no peer
Warning: nic virtio-net-pci.12 has no peer
Warning: nic virtio-net-pci.13 has no peer
Warning: nic virtio-net-pci.14 has no peer
Warning: nic virtio-net-pci.15 has no peer
Warning: nic virtio-net-pci.16 has no peer
Warning: nic virtio-net-pci.17 has no peer
Warning: nic virtio-net-pci.18 has no peer
Warning: nic virtio-net-pci.19 has no peer
Warning: nic virtio-net-pci.20 has no peer
Warning: nic virtio-net-pci.21 has no peer
Warning: nic virtio-net-pci.22 has no peer
Warning: nic virtio-net-pci.23 has no peer
qemu-system-x86_64: terminating on signal 15 from pid 13501
```

It is worth noting that if I start the same node right after GNS3 has been fired up, it starts without issue, in the exact same environment.

**Comment by jean-christophe-manciot** [comment]
If I try to stop the first running qemu node & then try to restart it, nothing works:
```
Server error from http://127.0.0.1:8000: IOSv_L2_10: Cannot connect to {}. Please check if GNS3 is allowed in your antivirus and firewall.
```
And the node log keeps being infinitely filled up, even when it's stopped, with:
```
e1000: Reading register at offset: 0x00002410. It is not fully implemented.
e1000: Reading register at offset: 0x00002418. It is not fully implemented.
e1000: Reading register at offset: 0x00002410. It is not fully implemented.
e1000: Reading register at offset: 0x00002418. It is not fully implemented.
...
```

**Comment by noplay** [comment]
How do you install GNS3 ? Package or from source?

**Comment by jean-christophe-manciot** [comment]
From source

**Comment by noplay** [comment]
Can you try pip3 install aiohttp==1.2.0 I suspect more recent version of
aiohttp to trigger a bug.

**Comment by jean-christophe-manciot** [comment]
I do have a more recent version:
```
# pip3 show aiohttp
Name: aiohttp
Version: 1.3.3
Summary: http client/server for asyncio
Home-page: https://github.com/KeepSafe/aiohttp/
Author: Andrew Svetlov <andrew.svetlov@gmail.com>, Nikolay Kim <fafhrd91@gmail.com>
Author-email: aio-libs@googlegroups.com
License: Apache 2
Location: /usr/local/lib/python3.5/dist-packages
Requires: async-timeout, chardet, multidict, yarl

# pip3 install aiohttp==1.2.0
Collecting aiohttp==1.2.0
  Using cached aiohttp-1.2.0-cp35-cp35m-manylinux1_x86_64.whl
Requirement already satisfied: chardet in /usr/local/lib/python3.5/dist-packages (from aiohttp==1.2.0)
Requirement already satisfied: yarl>=0.8.1 in /usr/local/lib/python3.5/dist-packages (from aiohttp==1.2.0)
Requirement already satisfied: async-timeout>=1.1.0 in /usr/local/lib/python3.5/dist-packages (from aiohttp==1.2.0)
Requirement already satisfied: multidict>=2.1.4 in /usr/local/lib/python3.5/dist-packages (from aiohttp==1.2.0)
Installing collected packages: aiohttp
  Found existing installation: aiohttp 1.3.3
    Uninstalling aiohttp-1.3.3:
      Successfully uninstalled aiohttp-1.3.3
Successfully installed aiohttp-1.2.0
```
Restarting GNS3 & 1 node, waiting 5 mn... and the issue is...  gone. Thanks.
Can you please update git-gns3-server/requirements.txt which contains ```aiohttp>=1.2.0```.

**Comment by noplay** [comment]
We can't change  the requirement this will break distribution  distributing aiohttp via their system package like Arch Linux. The Qt version provide by this distribution seem to have less trouble with Keep Alive

For ubuntu we recommend installation via the deb package it's more stable.

GNS3 2.0 contain a work around for this bug in Qt by disabling HTTP Keep-Alive for all clients  (the problem is not aiohttp, but a bug in Qt)

**Comment by ehlers** [comment]
See https://www.gns3.com/qa/i-updated-to-1-5-3-and-now-route

The same issue pops up, when you upgrade the GNS3 VM from 1.5.2 to 1.5.3. Then the latest aiohttp is installed (1.3.3) which doesn't work nicely with the 1.5.3 GNS3 client.

**Comment by noplay** [comment]
I pushed a fix for the VM in order to fix the version to 1.5 during the upgrade.
